securebanking-openbanking-uk-rs-simulator-server/src/main/java/com/forgerock/securebanking/openbanking/uk/rs/api/discovery/AvailableApiEndpointsResolver.java

```diff
@@ -66,12 +62,6 @@ public AvailableApiEndpointsResolver(RequestMappingHandlerMapping requestHandler
      * @return the {@link List} of {@link AvailableApiEndpoint} instances.
      */
     public List<AvailableApiEndpoint> getAvailableApiEndpoints() {
-        if (availableApis != null) {
-            return availableApis;
-        }
```

Comment:
Do not cache as (in theory) the X-Forwarded headers could vary between requests (or a client could initially forget to provide them).